x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/index.tsx

```diff
@@ -45,7 +45,10 @@ export function MetadataTable({ item, sections }: Props) {
       {filteredSections.map(section => {
         let sectionData: Record<string, unknown> = get(item, section.key);
         if (section.properties) {
-          sectionData = pick(item, section.properties);
+          sectionData = pick(sectionData, section.properties);
+          if (isEmpty(sectionData) && !section.required) {
+            return;
+          }
```

Comment:
We are doing the `required` check twice (here, and above where `filteredSections` is declared). Would it be possible to combine those two?

Reply:
@sqren I'm filtering out not required sections that don't have the properties listed. Even though I'd rather keep with this approach. I made some tests changing from `sections.filter` to `sections.reduce`, but the complexity increased.